src/modules/follows/FollowService.ts

```diff
@@ -0,0 +1,227 @@
+import { UserInfo } from "../../db/models/userInfo.model";
+import neo4jInstance, { QueryMode } from "../../db/neo4j/Neo4jInstance";
+import UserInfoService from "../../db/service/userInfo.service";
+import { followQueries } from "./FollowQueries";
+import UserInfoCollection from "../../db/models/userInfo.model";
+import mongoose from "mongoose";
+import neo4j from "neo4j-driver";
+
+class FollowService {
+	private userInfoService: UserInfoService;
+	constructor() {
+		this.userInfoService = new UserInfoService(UserInfoCollection);
+	}
+
+	/**
+	 * Set up relationship `Follows`
+	 * @param srcUserId ID of the actor
+	 * @param targetUserId ID of the target
+	 * @returns Returns error if `Follows` already exists
+	 */
+	async follow(srcUserId: UserInfo["userId"], targetUserId: UserInfo["userId"]) {
+		// Check if the relationship already exist or not
+		const relExists = await this.verifyRelFollows(srcUserId, targetUserId);
+		// If exists, return error and message: "Already exists"
+		if (relExists) {
+			return { success: false, message: "The actor already followed this user" };
+		}
+
+		// If does not exist, setup follow relationship between the two users
+		// Get the profile visibility of the target user
+		const targetUserMongoId = new mongoose.Types.ObjectId(targetUserId); // convert into MongoDB ID Object
+		const targerUserInfo = await this.userInfoService.findUserInfo({
+			userId: targetUserMongoId,
+		});
+		if (targerUserInfo?.isPrivate) {
+			// TODO: If private profile, continue here
+			// Requires: notification mechanism and set `Follows` relationship to `isPending`
+			return { success: false, message: "This function is not ready, work is in progress" };
+		} else {
+			return await this.createRelFollows(srcUserId, targetUserId);
```

Comment:
You don't need an await on a return statement
```suggestion
			return this.createRelFollows(srcUserId, targetUserId);
```